x/twap/logic_test.go

```diff
@@ -1596,3 +1597,57 @@ func geometricTestCaseFromDeltas0(startAccum, accumDiff sdk.Dec, timeDelta time.
 func geometricTestCaseFromDeltas1(startAccum, accumDiff sdk.Dec, timeDelta time.Duration, expectedTwap sdk.Dec) computeTwapTestCase {
 	return geometricTestCaseFromDeltas0(startAccum, accumDiff, timeDelta, sdk.OneDec().Quo(expectedTwap))
 }
+
+func (s *TestSuite) TestTwapLog() {
+	var expectedErrTolerance = osmomath.MustNewDecFromStr("0.000000000000000100")
+
+	tests := map[string]struct {
+		priceValue    osmomath.BigDec
+		expected      osmomath.BigDec
+		expectedPanic bool
+	}{
+		"Twaplog{912648174127941279170121098210.928219201902041311} = 99.525973560175362367": {
+			priceValue: osmomath.MustNewDecFromStr("912648174127941279170121098210.928219201902041311"),
+			// From: https://www.wolframalpha.com/input?i2d=true&i=log+base+2+of+912648174127941279170121098210.928219201902041311+with+20+digits
+			expected: osmomath.MustNewDecFromStr("99.525973560175362367"),
+		},
+	}
+	for name, test := range tests {
+		s.Run(name, func() {
+			osmoassert.ConditionalPanic(s.T(), test.expectedPanic, func() {
+				result := twap.TwapLog(test.priceValue.SDKDec())
+				result_by_customBaseLog := test.priceValue.CustomBaseLog(osmomath.BigDecFromSDKDec(twap.GeometricTwapMathBase))
+
+				s.Require().True(test.expected.Sub(osmomath.BigDecFromSDKDec(result)).Abs().LTE(expectedErrTolerance))
+				s.Require().True(result_by_customBaseLog.Sub(osmomath.BigDecFromSDKDec(result)).Abs().LTE(expectedErrTolerance))
+			})
+		})
+	}
+}
+
+func (s *TestSuite) TestTwapPow() {
+	var expectedErrTolerance = osmomath.MustNewDecFromStr("0.00000100")
+
+	tests := map[string]struct {
+		exponentValue osmomath.BigDec
+		expected      osmomath.BigDec
+		expectedPanic bool
+	}{
+		"TwapPow(0.5) = 1.41421356": {
+			exponentValue: osmomath.MustNewDecFromStr("0.5"),
+			//From: https://www.wolframalpha.com/input?i2d=true&i=power+base+2+exponent+0.5+with+9+digits
+			expected: osmomath.MustNewDecFromStr("1.41421356"),
+		},
```

Comment:
Same comment for removing table-driven boilerplate. The reason is that we are only keeping one test case